plane/src/drone/executor.rs

```diff
@@ -24,8 +25,14 @@ pub struct Executor<R: Runtime> {
 }
 
 impl<R: Runtime> Executor<R> {
-    pub fn new(runtime: Arc<R>, state_store: StateStore, ip: IpAddr) -> Self {
+    pub async fn new(runtime: Arc<R>, state_store: StateStore, ip: IpAddr) -> Self {
         let backends: Arc<DashMap<BackendName, Arc<BackendManager<R>>>> = Arc::default();
+        let state_store = Arc::new(Mutex::new(state_store));
+
+        Self::terminate_preexisting_backends(runtime.clone(), state_store.clone())
+            .await
+            .context("Failed to terminate all preexisting backends! Locks may be violated, Drone aborting startup.")
+            .unwrap();
```

Comment:
I'm ignoring this clippy comment -- the way I'm doing it with `.context().unwrap()` is a clean way to wrap the original error with additional context then panic. Open to other suggestions! Ultimately, I think Clippy simply doesn't understand the `.context()` and is simply reacting to the `.unwrap()`. A `.expect('bla')` by contrast will not include the original error.

Reply:
I think the "right thing" to do here would be to change the `Executor` interface to allow this to return a `Result`, but I'm fine with this for now until we know what `Err` type we would even want to use here.

I am surprised that `.expect()` doesn't include the error though, when I try it in the playground it does print the error: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=554e311a39e24d719c045ca0b6cd91db